### PR TITLE
Added SSL Support

### PR DESCRIPTION
**Pull request by tomcastleman**

### CODE REVIEW

File: classes/Kohana/Database/MySQLi.php [hunk 1]
```diff
@@ -49,7 +50,22 @@ public function connect()
...
 		{
-			$this->_connection = new mysqli($hostname, $username, $password, $database, $port, $socket);
+			if(is_array($ssl))
+			{
+				$this->_connection = mysqli_init();
+				$this->_connection->ssl_set(
+					Arr::get($ssl, 'ssl_client_key_file', ''),
+					Arr::get($ssl, 'ssl_client_cert_file', ''),
+					Arr::get($ssl, 'ca_cert_file', ''),
+					NULL,
```
Comment (shadowhand) [inline classes/Kohana/Database/MySQLi.php]: i think [all the options](http://php.net/manual/en/mysqli.ssl-set.php) should be available.

File: classes/Kohana/Database/MySQLi.php [hunk 2]
```diff
@@ -49,7 +50,22 @@ public function connect()
 
 		try
 		{
-			$this->_connection = new mysqli($hostname, $username, $password, $database, $port, $socket);
+			if(is_array($ssl))
+			{
+				$this->_connection = mysqli_init();
+				$this->_connection->ssl_set(
+					Arr::get($ssl, 'ssl_client_key_file', ''),
```
Comment (shadowhand) [inline classes/Kohana/Database/MySQLi.php]: according to the docs, `NULL` should be used instead of `''` for empty values.

**Comment by rjd22** [comment]
@tomcastleman is it possible to add test and documentation? You can find the docs in this repo under guide.

**Comment by tomcastleman** [comment]
@rjd22 OK, I will update the docs soon and add to this PR. Where / how do I add a test?

**Comment by rjd22** [comment]
@tomcastleman I see there is no test support for database yet sadly. Just the docs will be okay for now.

**Comment by tomcastleman** [comment]
@rjd22 How is that? No mention of MySQLi was there so I added it along with the SSL details.

**Comment by rjd22** [comment]
@tomcastleman awesome. Thanks.

**Pull request merged**